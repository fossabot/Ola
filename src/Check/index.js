import React from 'react'
import cx from 'classnames'
import {default as PT} from 'prop-types'

const Check = ({ type, children, extraClass, ...props }) => {
  return (
    <label className={cx('ola_check', extraClass)}>
      <input type={type} className="ola_check-input" {...props} />
      <div className="ola_check-label">
        <div className="ola_check-label-content">{ children }</div>
      </div>
    </label>
  )
}

Check.defaultProps = {
  type: 'checkbox',
  extraClass: null,
}

Check.propTypes = {
  /** Check Type */
  type: PT.oneOf(['checkbox', 'radio']),
  /** Extra className */
  extraClass: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default Check
