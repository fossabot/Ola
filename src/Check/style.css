.ola_check {
    padding: var(--size-3) 0;
    display: inline-block;

    &.is-area {
        display: block;
        padding: 0;

        & .ola_check-label {
            display: block;
            background: var(--white);
            border-radius: var(--radius-small);
            border: solid 2px var(--gray-light);
            padding: var(--size-4);
            transition: background-color 200ms;
        }

        & .ola_check-label:hover {
            border-color: var(--brand-dark);
        }

        & .ola_check-input:checked ~ .ola_check-label {
            border-color: var(--brand);
        }

        & .ola_check-input:active ~ .ola_check-label {
            border-color: var(--black);
        }

        & .ola_check-input:focus ~ .ola_check-label {
            box-shadow: 0 0 0 4px var(--brand-focus);
        }
    }

    &.is-option {
        display: block;
        padding: 0;

        & .ola_check-label {
            padding: var(--size-4);
            transition: background-color 200ms;
        }
        & .ola_check-input:checked ~ .ola_check-label {
            background-color: var(--gray-xlight);
        }
    }
}
.ola_check-input {
    position: absolute;
    opacity: 0;
}
.ola_check-label {
    font: var(--font-callout);
    display: flex;
    align-items: center;

    & img {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
        display: block;
        filter: grayscale(1);
        opacity: .7;
        transition-property: filter, opacity;
        transition-duration: var(--duration);
    }
}
.ola_check-label::before {
    --color: var(--gray-light);

    content: "";
    display: block;
    box-sizing: border-box;
    width: 24px;
    height: 24px;
    border: solid 2px;
    border-radius: var(--radius-small);
    flex: 0 0 auto;
    margin-right: 10px;
    color: var(--color);
    transition-property: color, background-color;
    transition-duration: var(--duration);
}
.ola_check-input:checked ~ .ola_check-label img {
    filter: none;
    opacity: 1;
}
.ola_check-input:checked ~ .ola_check-label::before {
    --color: var(--brand);
    background-color: currentColor;
    background-image: url('data:image/svg+xml;charset=UTF-8,<svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M17.919 7c.298 0 .553.109.764.326.211.217.317.48.317.786 0 .307-.107.57-.32.79l-7.56 7.777a1.02 1.02 0 0 1-.76.321 1.03 1.03 0 0 1-.768-.321l-4.28-4.394a1.09 1.09 0 0 1-.312-.79c0-.307.106-.57.317-.787.21-.217.466-.325.764-.325s.555.11.769.33l3.51 3.611 6.79-6.994c.214-.22.47-.33.769-.33z" fill="white"/></svg>');
    background-repeat: no-repeat;
    background-position: center center;
}
.ola_check-input:focus ~ .ola_check-label::before {
    --color: var(--brand);
    box-shadow: 0 0 0 4px var(--brand-focus);
}
.ola_check-input:hover ~ .ola_check-label::before {
    --color: var(--brand-dark);
}

/* Changes for radio */
.ola_check-input[type="radio"] {
    & ~ .ola_check-label::before {
        border-radius: 50%;
    }

    &:checked ~ .ola_check-label::before {
        background-image: url('data:image/svg+xml;charset=UTF-8,<svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><circle fill="white" cx="12" cy="12" r="4"/></svg>');
    }
}
