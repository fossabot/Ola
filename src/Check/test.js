import React from 'react'
import { Check } from './'
import renderer from 'react-test-renderer'

it('Default check is checkbox', () => {
  const tree = renderer
    .create(<Check>Accept terms</Check>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Check type checkbox', () => {
  const tree = renderer
    .create(<Check type="checkbox">Accept terms</Check>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Check type radio', () => {
  const tree = renderer
    .create(<Check type="radio">Accept terms</Check>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
