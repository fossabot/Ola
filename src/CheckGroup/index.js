import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const CheckGroup = ({ variant, children, className, ...props }) => {
  // Delete any error prop inherited from Field
  if(props.error) delete props['error'] // eslint-disable-line react/prop-types
  return (
    <div role="radiogroup" className={cx('ola_checkGroup', variant && `is-${variant}`, className)} { ...props }>
      { children }
    </div>
  )
}

CheckGroup.defaultProps = {
  variant: 'row',
  className: null,
}

CheckGroup.propTypes = {
  variant: PT.oneOf(['row', 'column', 'list']),
  /** Extra className */
  className: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export default CheckGroup
