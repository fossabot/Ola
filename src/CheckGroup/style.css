.ola_checkGroup {
    display: block;
    box-sizing: border-box;
    max-height: var(--size-11);
    width: 100%;
    border: solid 2px var(--gray-light);
    background: var(--white);
    border-radius: var(--radius-small);
    padding: var(--size-3) 0;
    overflow-y: auto;
    margin: var(--size-2) 0;

    & .ola_check {
        display: block;
        padding: 0;
    }
    & .ola_check-label {
        padding: var(--size-4);
        transition: background-color 200ms;
    }
    & .ola_check-input:checked ~ .ola_check-label {
        background-color: var(--gray-xlight);
    }
}
