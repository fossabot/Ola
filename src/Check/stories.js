import React from 'react'
import Check from './'

export default {
  title: 'Check',
  component: Check,
  args: {
    children: 'This is a check',
    checked: false
  }
}

export const Base = (args) => <Check {...args}>{args.children}</Check>
