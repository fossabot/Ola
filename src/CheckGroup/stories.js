import React from 'react'

import create from '../create'

import Field from '../Field'
import CheckGroup from './'

const random_option_values = [
  {
    label: 'Test 1',
    value: 'test1'
  },
  {
    label: 'Test 2',
    value: 'test2'
  },
  {
    label: 'Test 3',
    value: 'test3'
  },
  {
    label: 'Test 4',
    value: 'test4'
  },
  {
    label: 'Test 5',
    value: 'test5'
  },
  {
    label: 'Test 6',
    value: 'test6'
  },
  {
    label: 'Test 7',
    value: 'test7'
  },
  {
    label: 'Test 8',
    value: 'test8'
  },
  {
    label: 'Test 9',
    value: 'test9'
  },
  {
    label: 'Test 10',
    value: 'test10'
  },
  {
    label: 'Test 11',
    value: 'test11'
  }
]

create('CheckGroup')
  .add('Default', () => (
    <figure>
      <Field
        id="field-cutsom-checkGroup"
        label="Select the most accurate option:"
        hint="(optional)">
        <CheckGroup name="random" options={random_option_values} />
      </Field>
    </figure>
  ))
