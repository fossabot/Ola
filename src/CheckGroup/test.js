import React from 'react'
import renderer from 'react-test-renderer'
import CheckGroup from './'
import Check from '../Check'

const options = [
  {
    label: 'Option 1',
    value: 1,
  },
  {
    label: 'Option 2',
    value: 2,
  },
  {
    label: <strong>Option 3</strong>,
    value: 3,
  }
]

it('Default checkGroup', () => {
  const tree = renderer
    .create(
      <CheckGroup>
        { options.map((option, key) => <Check key={key} variant="area" value={option.value}>{option.label}</Check>)}
      </CheckGroup>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Checkgroup in columns', () => {
  const tree = renderer
    .create(
      <CheckGroup variant="column">
        { options.map((option, key) => <Check key={key} variant="area" value={option.value}>{option.label}</Check>)}
      </CheckGroup>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Checkgroup in list', () => {
  const tree = renderer
    .create(
      <CheckGroup variant="list">
        { options.map((option, key) => <Check key={key} variant="option" value={option.value}>{option.label}</Check>)}
      </CheckGroup>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
