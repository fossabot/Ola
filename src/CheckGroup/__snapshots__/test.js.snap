// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default CheckGroup 1`] = `
<div
  className="ola_checkGroup"
  onChange={[Function]}
  role="radiogroup"
>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name={null}
      type="radio"
      value={1}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        Option 1
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name={null}
      type="radio"
      value={2}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        Option 2
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={null}
      className="ola_check-input"
      name={null}
      type="radio"
      value={3}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        Option 3
      </div>
    </div>
  </label>
</div>
`;

exports[`Default CheckGroup option checked 1`] = `
<div
  className="ola_checkGroup"
  onChange={[Function]}
  role="radiogroup"
>
  <label
    className="ola_check"
  >
    <input
      checked={true}
      className="ola_check-input"
      name={null}
      type="radio"
      value={1}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        Option 1
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={false}
      className="ola_check-input"
      name={null}
      type="radio"
      value={2}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        Option 2
      </div>
    </div>
  </label>
  <label
    className="ola_check"
  >
    <input
      checked={false}
      className="ola_check-input"
      name={null}
      type="radio"
      value={3}
    />
    <div
      className="ola_check-label"
    >
      <div
        className="ola_check-label-content"
      >
        Option 3
      </div>
    </div>
  </label>
</div>
`;
